timemachine/md/minimizer.py

```diff
@@ -140,85 +142,181 @@ def minimize_host_4d(
     ff: ff.Forcefield
         Wrapper class around a list of handlers
 
-    mol_coords: list of np.ndarray
+    mol_coords: list of np.ndarray, optional
         Pre-specify a list of mol coords. Else use the mol.GetConformer(0)
 
-    windows: integer
-        Number of lambda windows to lower the mols into the host via 4D decoupling
+    minimizer_steps_per_window: integer
+        Number of steps to run each window of the FIRE minimizer with
 
-    n_steps_per_window: integer
-        Number of steps to evaluate at each window
+    minimizer_windows: integer
+        Number of windows to run FIRE minimizer over
+
+    equilibration_steps: integer
+        Number of steps to run MD with
+
+    pressure: float
+        Pressure to run barostat with
+
+    temperature: float
+        Temperature to run MD with
+
+    barostat_interval: integer
+        How often to run the barostat
+
+    seed: integer
+        Seed for barostat and integrator
 
     Returns
     -------
-    np.ndarray
-        This returns minimized host_coords.
+    2-tuple of host coordinates and box
+        the minimized host_coords and the new box
 
     """
     box = host_config.box
     assert box.shape == (3, 3)
 
+    minimized_host_coords = fire_minimize_host(
+        mols,
+        host_config,
+        ff,
+        mol_coords=mol_coords,
+        n_windows=minimizer_windows,
+        n_steps_per_window=minimizer_steps_per_window,
+    )
+
     host_bps, host_masses = openmm_deserializer.deserialize_system(host_config.omm_system, cutoff=1.2)
 
     num_host_atoms = host_config.conf.shape[0]
 
     if len(mols) == 1:
         top = topology.BaseTopology(mols[0], ff)
     elif len(mols) == 2:
-        top = topology.DualTopologyMinimization(mols[0], mols[1], ff)
+        top = topology.DualTopology(mols[0], mols[1], ff)
```

Comment:
curiosity/oddity: this PR changes the topology in the function `minimize_host_4d` from `DualTopologyMinimization` to `DualTopology`, but changes the topology in the function `make_host_du_dx_fxn` from `DualTopology` to `DualTopologyMinimization`.

(Unclear if these are neutral mutations vs. functionally important)

Reply:
They are pretty much interchangable, will stick with `DualTopologyMinimization` here to avoid confusion.

The other difference is that `DualTopology` ignores lambda and `DualTopologyMinimization` adjusts the w coord to cutoff at lamb = 1.0 and at lamb = 0.0 the w coord is zero.

Reply:
Stick to using `DualTopologyMinimization` in https://github.com/proteneer/timemachine/pull/1339/commits/1699559316057dac6222b5686d34e6755e22c2ff

> changes the topology in the function make_host_du_dx_fxn from DualTopology to DualTopologyMinimization

Makes it possible to modify the cutoff value and use the function to run either barker/fire with this minimization approach.